### PR TITLE
Import smimecosign

### PR DESCRIPTION
**Pull request by wlynch**
#### Summary

This PR imports and renames [smimecosign](https://github.com/wlynch/smimecosign) to gitsign, preserving the origin repo commit history. 

#### Ticket Link

n/a

#### Release Note

```release-note
Initial import of gitsign! Provides a git utility for signing git commits with ephemeral sigstore keys.
```

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        git config --global user.email "test@example.com"
+        git config --global user.name "gitsign"
+        git config --global gpg.x509.program gitsign
+        git config --global gpg.format x509
+        git config --global commit.gpgsign true
+
+        # Verify tool is on our path
+        gitsign -h
+
+        # Sign commit
```
Comment (imjasonh) [inline .github/workflows/go.yml:52]:
```suggestion
        # Sign and verify commit
```

File: README.md [hunk 2]
```diff
@@ -1,6 +1,351 @@
...
+```sh
+$ git config --global gpg.x509.program gitsign
+$ git config --global gpg.format x509
+```
+
+### Cosign configuration options
+
+| Environment Variable  | Default                          | Description                                |
+| --------------------- | -------------------------------- | ------------------------------------------ |
+| COSIGN_FULCIO_URL     | https://fulcio.sigstore.dev      | Address of Fulcio server                   |
```
Comment (imjasonh) [inline README.md:35]: https://github.com/sigstore/gitsign/issues/2

File: git/git.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/pkg/errors"
+	"github.com/sigstore/cosign/cmd/cosign/cli/fulcio/fulcioroots"
+	"github.com/sigstore/gitsign/fulcio"
+	"github.com/sigstore/gitsign/rekor"
+	"github.com/sigstore/gitsign/signature"
+	"github.com/sigstore/rekor/pkg/generated/models"
+)
+
+func Sign(ctx context.Context, ident *fulcio.Identity, data []byte, opts signature.SignOptions) ([]byte, *x509.Certificate, error) {
```
Comment (imjasonh) [inline git/git.go:20]: https://github.com/sigstore/gitsign/issues/3

File: fulcio/fulcio.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	"github.com/sigstore/cosign/cmd/cosign/cli/sign"
+	"github.com/sigstore/cosign/pkg/providers"
+)
+
+type Identity struct {
+	sv     *sign.SignerVerifier
+	stderr io.Writer
+}
+
+func NewIdentity(ctx context.Context, w io.Writer) (*Identity, error) {
```
Comment (imjasonh) [inline fulcio/fulcio.go:37]: https://github.com/sigstore/gitsign/issues/3

**Review by imjasonh — APPROVED** [review]
🎉

File: .github/workflows/go.yml (hunk 1 above)
Comment (wlynch) [reply .github/workflows/go.yml:52]: Want to keep diffs largely unchanged for this pass - will follow up with another PR for this!

**Pull request merged**